pgbedrock/core_configure.py

```diff
@@ -131,22 +138,115 @@ def run_password_sql(cursor, all_password_sql_to_run):
         common.fail(msg=common.FAILED_QUERY_MSG.format(query, e))
 
 
-def verify_spec(spec):
-    """ Ensure keywords make sense, particularly because it's easy to use singulars
-    instead of plurals (e.g. 'table' instead of 'tables') """
+def verify_schema(spec):
+    """Checks spec for schema errors."""
     assert isinstance(spec, dict)
+    error_messages = []
 
     schema = yaml.load(SPEC_SCHEMA_YAML)
     v = cerberus.Validator(schema)
-    error_messages = []
     for rolename, config in spec.items():
         if not config:
             continue
-
         v.validate(config)
         for field, err_msg in v.errors.items():
             error_messages.append(VALIDATION_ERR_MSG.format(rolename, field, err_msg[0]))
 
+    return error_messages
+
+
+def check_for_multi_schema_owners(spec):
+    """Checks spec for schema with multiple owners."""
+    assert isinstance(spec, dict)
+    error_messages = []
+
+    globally_owned_schemas = defaultdict(list)
+    for role, config in spec.items():
+        if not config:
+            continue
+        if config.get('has_personal_schema'):
+            # indicates a role has a personal schema with its same name
+            globally_owned_schemas[role].append(role)
+        if config.get('owns') and config['owns'].get('schemas'):
+            role_owned_schemas = config['owns']['schemas']
+            for schema in role_owned_schemas:
+                globally_owned_schemas[schema].append(role)
+    for schema, owners in globally_owned_schemas.items():
+        if len(owners) > 1:
+            owners_formatted = ", ".join(owners)
+            error_messages.append(MULTIPLE_SCHEMA_OWNER_ERR_MSG.format(schema, owners_formatted))
+
+    return error_messages
+
+
+def check_read_write_obj_references(spec):
+    """Verifies objects aren't defined in both read and write privileges sections."""
+    assert isinstance(spec, dict)
```

Comment:
Technically this line isn't necessary since the `verify_schema()` function has this same line.